src/components/frame/v5/pages/BalancePage/partials/BalanceTable/hooks.tsx

```diff
@@ -171,3 +234,137 @@ export const useGetTableMenuProps = () =>
+        );
+
+        const formattedValue = getFormattedNumeralValue(
+          convertedValue,
+          currentTokenBalance,
+        );
+
+        return {
+          ...item,
+          balance: typeof formattedValue === 'string' ? formattedValue : '',
+        };
+      }),
+    [colony, balances, selectedTeam],
+  );
+
+  const sortedTokens = useMemo(
+    () =>
+      tokensData?.sort((a, b) => {
+        if (!a.balance || !b.balance) return 0;
+        return parseInt(b.balance, 10) - parseInt(a.balance, 10);
+      }),
+    [tokensData],
+  );
+
+  const filters: FiltersProps<BalanceTableFilters> = {
+    onChange: setValue,
+    onSearch: setSearchValue,
+    searchValue,
+    value,
+    customLabel: formatText(MSG.customLabel),
+    items: [
+      {
+        name: 'type',
+        label: formatText(MSG.filterTokenType),
+        title: formatText(MSG.filterApprovedTokens),
+        icon: 'coin-vertical',
+        items: sortedTokens
+          ? sortedTokens.map(({ token }) => ({
+              name: token?.tokenAddress || '',
+              label: token && (
+                <div className="flex items-center gap-2">
+                  <TokenIcon token={token} size="xxxs" />
+                  {token.symbol}
+                </div>
+              ),
+            }))
+          : [],
+      },
+      {
+        name: 'attribute',
+        label: formatText(MSG.filterAttributes),
+        title: formatText(MSG.filterAttributeTypes),
+        icon: 'folder-simple-lock',
+        items: [
+          {
+            name: 'native',
+            label: formatText(MSG.typeNative),
+          },
+          {
+            name: 'reputation',
+            label: formatText(MSG.typeReputation),
+          },
+        ],
+      },
+    ],
+  };
+
+  const tokens: TokensProps[] | undefined = sortedTokens?.map((token) => ({
+    ...token,
+    isApproved: sortedTokens?.some(
+      ({ token: colonyTokens }) =>
+        colonyTokens?.tokenAddress === token?.token?.tokenAddress,
+    ),
+    isNative: token.token?.tokenAddress === nativeToken?.tokenAddress,
+  }));
+
+  const visibleTokens: TokensProps[] | undefined = tokens?.filter(
+    (visibleToken): TokensProps | boolean | undefined => {
+      const { type = {}, attribute } = value;
+
+      if (Object.values(type).some((tokenTypeFilter) => tokenTypeFilter)) {
+        return type[visibleToken?.token?.tokenAddress || 0];
+      }
+
+      if (
+        Object.values(attribute || {}).some(
+          (attributeTypeFilter) => attributeTypeFilter,
+        )
+      ) {
+        return (
+          (attribute?.native && visibleToken.isNative) ||
+          (attribute?.reputation && !visibleToken.isNative)
+        );
+      }
+
+      return visibleToken;
+    },
+  );
+
+  const searchedTokens = visibleTokens?.filter(
+    ({ token }) =>
+      token?.name.toLowerCase().includes(searchValue.toLowerCase()) ||
+      token?.symbol.toLowerCase().includes(searchValue.toLowerCase()),
+  );
```

Comment:
It'd be nice to extract all the filtering logic into helper functions to keep this clean, e.g.:
```ts
const filteredTokens = tokens
  ?.filter(filterTokensByAttribute)
  .filter(filterTokensBySearchValue)
```